index.js

```diff
@@ -88,6 +89,21 @@ simpleGet.concat = function (opts, cb) {
   }
 })
 
+function setReferrer (opts) {
+  var key = Object.keys(opts.headers).find(function (h) {
+    h = h.toLowerCase()
+    return h === 'referer' || h === 'referrer'
+  }) || 'referer'
+
+  opts.headers = opts.headers || {}
+  opts.headers[key] = url.format({
+    hostname: opts.hostname,
+    port: opts.port,
+    protocol: opts.protocol,
+    pathname: opts.path
+  })
```

Comment:
I'm not sure it actually makes sense to set the `referer` header to the previous URL. That doesn't match what browsers do. See http://stackoverflow.com/a/5441932

The `referer` header really ought to be the url of the page that the initial request is made from. Not the url of the previous page in the redirect chain. That make sense?